apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -152,6 +247,24 @@ type Patch struct {
 	// input to be transformed.
 	// +optional
 	Transforms []Transform `json:"transforms,omitempty"`
+
+	// Type can be set to refer to a named patch defined against the parent
+	// composition. If a Patch has a type set, then all other fields will be
+	// ignored and the referenced patch will be used instead.
+	// +optional
+	Type PatchType `json:"type,omitempty"`
+
+	// ID can be set on a patch of type ref to refer to a specific patch set by
+	// name.
+	// +optional
+	ID string `json:"id,omitempty"`
+}
+
+// SetDefaults makes sure patches have the correct default values applied.
+func (c *Patch) SetDefaults() {
+	if c.Type == "" {
+		c.Type = PatchTypeStandard
+	}
 }
```

Comment:
We recently updated the project to use v1 CRDs, which support default values as part of the CRD schema. It seems like kubebuilder has undocumented support for this, which may allow us to avoid this `SetDefaults` method. See https://github.com/kubernetes-sigs/controller-tools/pull/337/files